ara/cli/prometheus.py

```diff
@@ -0,0 +1,282 @@
+            "range": Gauge("ara_hosts_range", "Limit metric collection to the N most recent hosts"),
+            "skipped": Gauge("ara_hosts_skipped", "Number of skipped tasks on a host", labels),
+            "total": Gauge("ara_hosts_total", "Hosts recorded by ara"),
+            "unreachable": Gauge("ara_hosts_unreachable", "Number of unreachable errors on a host", labels),
+        }
+        self.metrics["range"].set(self.limit)
+
+    def collect_metrics(self, created_after=None):
+        hosts = get_search_results(self.client, "hosts", self.limit, created_after)
+        # Save the most recent timestamp so we only scrape beyond it next time
+        if hosts:
+            created_after = cli_utils.increment_timestamp(hosts[0]["created"])
+            self.log.info(f"updating metrics for {len(hosts)} hosts...")
+
+        for host in hosts:
+            self.metrics["total"].inc()
+
+            # Gather the values of each label so we can attach them to our metrics
+            labels = {label: host[label] for label in self.labels}
+
+            # The values of "changed", "failed" and so on are integers so we can
+            # use them as values for our metric
+            for status in ["changed", "failed", "ok", "skipped", "unreachable"]:
+                if host[status]:
+                    self.metrics[status].labels(**labels).set(host[status])
+
+        return created_after
+
+
+class PrometheusExporter(Command):
+    """Exposes a prometheus exporter to provide metrics from an instance of ara"""
+
+    log = logging.getLogger(__name__)
+
+    def get_parser(self, prog_name):
+        parser = super().get_parser(prog_name)
+        parser = global_arguments(parser)
+        # fmt: off
+        parser.add_argument(
+            '--playbook-limit',
+            help='Max number of playbooks to request at once (default: 1000)',
+            default=1000,
+            type=int
+        )
+        parser.add_argument(
+            '--task-limit',
+            help='Max number of tasks to request at once (default: 2500)',
+            default=2500,
+            type=int
+        )
+        parser.add_argument(
+            '--host-limit',
+            help='Max number of hosts to request at once (default: 2500)',
+            default=2500,
+            type=int
+        )
+        parser.add_argument(
+            '--poll-frequency',
+            help='Seconds to wait until querying ara for new metrics (default: 60)',
+            default=60,
+            type=int
+        )
+        parser.add_argument(
+            '--prometheus-port',
+            help='Port on which the prometheus exporter will listen (default: 8001)',
+            default=8001,
+            type=int
+        )
+        parser.add_argument(
+            '--max-days',
+            help='Maximum number of days to backfill metrics for (default: 90)',
+            default=90,
+            type=int
+        )
```

Comment:
I think it could be interesting for the exporter to be able to filter queries like the general CLI commands work, for example ``ara playbook list`` ([docs](https://ara.readthedocs.io/en/latest/cli.html#ara-playbook-list)) has:
```
  --ansible_version <ansible_version>
                        List playbooks that ran with the specified Ansible
                        version (full or partial)
  --client_version <client_version>
                        List playbooks that were recorded with the specified
                        ara client version (full or partial)
  --server_version <server_version>
                        List playbooks that were recorded with the specified
                        ara server version (full or partial)
  --python_version <python_version>
                        List playbooks that were recorded with the specified
                        python version (full or partial)
  --user <user>         List playbooks that were run by the specified user
                        (full or partial)
  --controller <controller>
                        List playbooks that ran from the provided controller
                        (full or partial)
  --name <name>         List playbooks matching the provided name (full or
                        partial)
  --path <path>         List playbooks matching the provided path (full or
                        partial)
  --status <status>     List playbooks matching a specific status
                        ('completed', 'running', 'failed')
```